expr/pool.go

```diff
@@ -12,3 +16,17 @@ var pointSlicePool *sync.Pool
 func Pool(p *sync.Pool) {
 	pointSlicePool = p
 }
+
+// pointSlicePoolGet returns a pointslice of at least minCap capacity.
+// similar code lives also in api.Fix(). at some point we should really clean up our pool code.
+func pointSlicePoolGet(minCap int) []schema.Point {
+	// 3 attempts to get a sufficiently sized slice from the pool. if it fails, allocate a new one
+	for attempt := 1; attempt < 4; attempt++ {
+		candidate := pointSlicePool.Get().([]schema.Point)
+		if cap(candidate) >= minCap {
+			return candidate
+		}
+		pointSlicePool.Put(candidate)
```

Comment:
> So if we Get/Put/Get/Put/Get/Put it's highly likely that we will be Get'ing and Put'ing the same candidate over and over again.

Something we could verify experimentally. But I rather not rely on specific behavior that may evolve over time. Especially since the cost of doing a few put/gets should be negligible 

> I couldn't verify it was actually helping.

bummer. would be interesting to collect some stats on the pool in terms of hit vs miss rate, excess capacity, etc.

Reply:
> But I rather not rely on specific behavior that may evolve over time.

Why have extra code that is almost certainly not going to provide any benefit at the time of writing and no evidence that it would provide any benefit in the future?

> would be interesting to collect some stats on the pool in terms of hit vs miss rate, excess capacity, etc.

In my size-tiered pool tests I did add hit/miss stats. I was a little worried that the measurement might be a bit expensive, but I doubt it was. I'd rather start with this than have the multi-attempt code, TBH.

Reply:
ok. have a look at my last commit